protostuff-runtime/src/test/java/io/protostuff/runtime/AbstractRuntimeCollectionSchemaTest.java

```diff
@@ -782,5 +782,255 @@ public void testCollectionAndMapSchema() throws Exception
 
         roundTrip(p, schema, pipeSchema);
     }
+    
+    @SuppressWarnings("serial")
+    static final class PersistentObjectList<T> extends ArrayList<T>
+    {
+        public final int id;
+        
+        public PersistentObjectList()
+        {
+            id = 0;
+        }
+        
+        public PersistentObjectList(int id)
+        {
+            this.id = id;
+        }
+    }
+    
+    
+    @SuppressWarnings("serial")
+    static final class PersistentObjectMap<K,V> extends HashMap<K,V>
+    {
+        public final int id;
+        
+        public PersistentObjectMap()
+        {
+            id = 0;
+        }
+        
+        public PersistentObjectMap(int id)
+        {
+            this.id = id;
+        }
+    }
+    
+    static final class ObjectWrapper
+    {
+        Object obj;
+    }
+    
+    static final class ListWrapper
+    {
+        List<?> obj;
+    }
+    
+    static final class MapWrapper
+    {
+        Map<?,?> obj;
+    }
+    
+    /*static final class EagerDelegate<T> implements Delegate<T>
```

Comment:
Its a placeholder for the time when it will be actually added.  Registering that for a specific message/pojo overrides the merge functionality (e.g. message not merged).